From f9d49232277529d07e7cf30bd59328b40a44497c Mon Sep 17 00:00:00 2001
From: Varun Sethi <Varun.Sethi@freescale.com>
Date: Wed, 21 Aug 2013 18:59:54 +0530
Subject: [PATCH 137/383] Update the operation mapping settings required by
 the DSP side for Maple and DMA.

For the DSP side the basic requirement for stashing is to prevent device
initiated reads and writes going to the DDR. RWNITC and WWSOT operation
mappings aid in achieving these objectives.

Signed-off-by: Varun Sethi <Varun.Sethi@freescale.com>
Change-Id: I1634ff877d9c8e7c6a3de95aee4a3848cc149973
Reviewed-on: http://git.am.freescale.net:8181/4128
Reviewed-by: Rivera Jose-B46482 <German.Rivera@freescale.com>
Tested-by: Rivera Jose-B46482 <German.Rivera@freescale.com>
[Original patch taken from QorIQ-SDK-V1.5-SOURCE-20131219-yocto.iso]
Signed-off-by: Bin Jiang <bin.jiang@windriver.com>
---
 drivers/iommu/fsl_pamu.c |   26 ++++++--------------------
 include/linux/iommu.h    |    4 +---
 2 files changed, 7 insertions(+), 23 deletions(-)

diff --git a/drivers/iommu/fsl_pamu.c b/drivers/iommu/fsl_pamu.c
index 3519a2f..e0d5b8e 100644
--- a/drivers/iommu/fsl_pamu.c
+++ b/drivers/iommu/fsl_pamu.c
@@ -778,26 +778,12 @@ static void __init setup_omt(struct ome *omt)
 	ome->moe[IOE_DIRECT0_IDX] = EOE_LDEC | EOE_VALID;
 	ome->moe[IOE_DIRECT1_IDX] = EOE_LDEC | EOE_VALID;
 
-	/* Configure OMI_DMA */
-	ome = &omt[OMI_DMA];
-	ome->moe[IOE_READ_IDX]  = EOE_VALID | EOE_RSA;
-	ome->moe[IOE_EREAD0_IDX] = EOE_VALID | EOE_RSA;
-	ome->moe[IOE_WRITE_IDX] = EOE_VALID | EOE_WWSA;
-	ome->moe[IOE_EWRITE0_IDX] = EOE_VALID | EOE_WWSA;
-
-	/* Configure OMI_DMA_READI */
-	ome = &omt[OMI_DMA_READI];
-	ome->moe[IOE_READ_IDX]  = EOE_VALID | EOE_READI;
-	ome->moe[IOE_EREAD0_IDX] = EOE_VALID | EOE_READI;
-	ome->moe[IOE_WRITE_IDX] = EOE_VALID | EOE_WWSA;
-	ome->moe[IOE_EWRITE0_IDX] = EOE_VALID | EOE_WWSA;
-
-	/* Configure OMI_MAPLE */
-	ome = &omt[OMI_MAPLE];
-	ome->moe[IOE_READ_IDX]  = EOE_VALID | EOE_RSA;
-	ome->moe[IOE_EREAD0_IDX] = EOE_VALID | EOE_RSA;
-	ome->moe[IOE_WRITE_IDX] = EOE_VALID | EOE_WWSA;
-	ome->moe[IOE_EWRITE0_IDX] = EOE_VALID | EOE_WWSA;
+	/* Configure OMI_DSP */
+	ome = &omt[OMI_DSP];
+	ome->moe[IOE_READ_IDX]  = EOE_VALID | EOE_RWNITC;
+	ome->moe[IOE_EREAD0_IDX] = EOE_VALID | EOE_RWNITC;
+	ome->moe[IOE_WRITE_IDX] = EOE_VALID | EOE_WWSAO;
+	ome->moe[IOE_EWRITE0_IDX] = EOE_VALID | EOE_WWSAO;
 }
 
 /*
diff --git a/include/linux/iommu.h b/include/linux/iommu.h
index e08c9c3..b39549a 100644
--- a/include/linux/iommu.h
+++ b/include/linux/iommu.h
@@ -91,9 +91,7 @@ enum omap_index {
 	OMI_QMAN_PRIV,
 	OMI_CAAM,
 	OMI_PMAN,
-	OMI_DMA,
-	OMI_DMA_READI,
-	OMI_MAPLE,
+	OMI_DSP,
 	OMI_MAX,
 };
 
-- 
1.7.5.4

